internal/opts/config.go

```diff
@@ -0,0 +1,192 @@
+package opts
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"time"
+
+	flag "github.com/spf13/pflag"
+
+	"github.com/spf13/viper"
+)
+
+type Config struct {
+
+	// region level configuration.
```

Comment:
We can split this struct into multiple smaller structs under a single struct.

Reply:
Possible, but didn't want to break the config signature in this PR. This config just moves all flags into this struct. Splitting this can be taken up separately. (Even the discovery-ini should be moved into a struct and refactored)

Reply:
Created https://github.com/flipkart-incubator/dkv/issues/129